File: app/src/main/java/com/sharpdroid/registroelettronico/Utils/MyLinkedMap.java
package com.sharpdroid.registroelettronico.Utils;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Set;

public class MyLinkedMap<K, V> extends LinkedHashMap<K, V> {

    public V getValue(int i) {

        Map.Entry<K, V> entry = this.getEntry(i);
        if (entry == null) return null;

        return entry.getValue();
    }

    public Map.Entry<K, V> getEntry(int i) {
        // check if negetive index provided
        Set<Map.Entry<K, V>> entries = entrySet();
        int j = 0;

        for (Map.Entry<K, V> entry : entries)
            if (j++ == i) return entry;

        return null;

    }

    public K getKey(int i) {
        Map.Entry<K, V> entry = this.getEntry(i);
        if (entry == null) return null;

        return entry.getKey();
    }


}

File: app/src/main/java/com/sharpdroid/registroelettronico/Utils/MyLinkedMap.kt
package com.sharpdroid.registroelettronico.Utils

import java.util.*

class MyLinkedMap<K, V> : LinkedHashMap<K, V>() {

    fun getValue(i: Int): V? {

        val entry = this.getEntry(i) ?: return null

        return entry.value
    }

    fun getEntry(i: Int): Map.Entry<K, V>? {
        // check if negetive index provided
        val entries = entries
        var j = 0

        for (entry in entries)
            if (j++ == i) return entry

        return null

    }

    fun getKey(i: Int): K? {
        val entry = this.getEntry(i) ?: return null

        return entry.key
    }


}